slack-conversation.c

```diff
@@ -255,15 +290,28 @@ void slack_get_history(SlackAccount *sa, SlackObject *conv, const char *since, u
 		return;
 	}
 
+	struct get_history *h = g_new(struct get_history, 1);
+	h->conv = g_object_ref(conv);
+	h->since = g_strdup(since);
+	h->thread = (thread_ts != NULL);
+
 	char count_buf[6] = "";
-	snprintf(count_buf, 5, "%u", MIN(count,500));
-	slack_api_get(sa, get_history_cb, conv, "conversations.history", "channel", id, "oldest", since ?: "0", "limit", count_buf, NULL);
+	snprintf(count_buf, 5, "%u", MIN(count, SLACK_HISTORY_LIMIT_NUM));
+	if (thread_ts)
+		slack_api_get(sa, get_history_cb, h, "conversations.replies", "channel", id, "limit", count_buf, "ts", thread_ts, NULL);
+	else
+		slack_api_get(sa, get_history_cb, h, "conversations.history", "channel", id, "limit", count_buf, NULL);
 }
 
 void slack_get_history_unread(SlackAccount *sa, SlackObject *conv, json_value *json) {
 	slack_get_history(sa, conv,
 			json_get_prop_strptr(json, "last_read"),
-			json_get_prop_val(json, "unread_count", integer, -1));
+			SLACK_HISTORY_LIMIT_NUM,
```

Comment:
Is there some reason not to use "unread_count" here if it's available?

Reply:
This is described in the original commit message for threads, but it is indeed not very discoverable. I copied that message into a comment here, to make it clearer. I encourage you to read it.

But it's good that you pointed it out, because I also realized we can optimize this: It's only needed when thread replies are turned on, so I went ahead and made this conditional.